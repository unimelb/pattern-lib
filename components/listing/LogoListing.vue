<template>
  <section-wrap>
    <div class="logo-listing">
      <a
        v-for="(item, index) in data"
        :key="index"
        :href="item.url"
        class="logo-listing__link">
        <img
          :alt="item.alt"
          :src="item.src"
          class="logo-listing__image"
        >
      </a>
    </div>
  </section-wrap>
</template>

<script>
export default {
  props: {
    data: {
      type: Array,
      default: () => [],
    },
  },
};
</script>
