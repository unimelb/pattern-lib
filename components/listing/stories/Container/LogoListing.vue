<template>
  <LogoListing :data="sample" />
</template>

<script>
import LogoListing from '../../LogoListing.vue';
import sample from '../../sample.json';

export default {
  components: { LogoListing },
  readme: { source: true, html: false },
  data() {
    return {
      sample,
    };
  },
};
</script>
