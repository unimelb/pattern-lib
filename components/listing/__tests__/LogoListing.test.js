import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import LogoListing from '../LogoListing.vue';


expect.extend(toHaveNoViolations);

describe('LogoListing', () => {
  it('should match snapshot', () => {
    const result = shallow(LogoListing).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(LogoListing).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
