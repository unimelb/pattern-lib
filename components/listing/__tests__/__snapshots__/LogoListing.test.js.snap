// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoListing should match snapshot 1`] = `
<section-wrap>
  <div
    class="logo-listing"
  />
</section-wrap>
`;
